### DISCUSSION TITLE
第4节课作业(19班)

### DISCUSSION THREAD

**Discussion by LYMDLUT** [InternLM/Tutorial#251] [作业]
第4节课作业(19班)

**Comment by hu-qi** [comment]
- 基础作业：

构建数据集，使用 XTuner 微调 InternLM-Chat-7B 模型, 让模型学习到它是你的智能小助手，效果如下图所示，本作业训练出来的模型的输出需要将不要葱姜蒜大佬替换成自己名字或昵称

实际训练中，不到1个epoch就能达到效果了，2个epoch基本就稳了，感觉有点过拟合了。

![image](https://github.com/InternLM/tutorial/assets/17986122/9ada9efa-0655-4c83-9744-f8257324d5c0)

![image](https://github.com/InternLM/tutorial/assets/17986122/28d0c2be-7528-41e4-8576-f62a8f9292c9)





- 进阶作业：

将训练好的Adapter模型权重上传到 OpenXLab、Hugging Face 或者 MoelScope 任一一平台。

https://openxlab.org.cn/models/detail/huqi/I-nternLMForHuqi

~将训练好后的模型应用部署到 OpenXLab 平台，参考部署文档请访问：https://aicarrier.feishu.cn/docx/MQH6dygcKolG37x0ekcc4oZhnCe~

**Comment by romanwin28** [comment]
基础作业：
<img width="915" alt="微信截图_20240113190819" src="https://github.com/InternLM/tutorial/assets/146464537/3e4289ad-55b7-45f9-8360-1a8458e0419a">

<img width="904" alt="微信截图_20240113203041" src="https://github.com/InternLM/tutorial/assets/146464537/985d2628-ed6e-4d46-ab01-0254521621e5">

**Comment by Jiangzzzzl** [comment]
<img width="1241" alt="159b5c8f739cccc90fe092e454f4f37" src="https://github.com/InternLM/tutorial/assets/150671792/dc856312-ba57-42dd-835b-d7896c8c5bf3">

**Comment by Sixdes** [comment]
https://github.com/Sixdes/puyu-course/tree/main/homework0114

**Comment by JankinTian** [comment]
https://blog.csdn.net/xiao_xian_/article/details/135612476

**Comment by jts250** [comment]
https://github.com/jts250/homework_llm/blob/main/homework4.pdf

**Comment by melonwine** [comment]
https://github.com/melonwine/InternLMPlay/blob/master/04/asgmt04.md

**Comment by AllYoung** [comment]
https://blog.csdn.net/sunshine_youngforyou/article/details/135637778

**Comment by mkdirmushroom** [comment]
[第4节课作业](https://github.com/mkdirmushroom/note/blob/main/InternLM/tutorial3_assignment.md)

**Comment by Hamasuta666** [comment]
第4节课笔记（作业在文末）：https://zhuanlan.zhihu.com/p/680582728
